python/pyarrow/error.pyx

```diff
@@ -19,13 +19,52 @@ from pyarrow.includes.libarrow cimport CStatus
 from pyarrow.includes.common cimport c_string
 from pyarrow.compat import frombytes
 
+
 class ArrowException(Exception):
     pass
 
+
+class ArrowInvalid(ValueError, ArrowException):
```

Comment:
Is it an antipattern? I have never thought there was anything wrong either with

* deriving from built-in exceptions, like ValueError
* raising built-in exceptions directly (pretty much all Python data libraries -- pandas, NumPy, scikit-learn, etc. already do this)

So there's a couple ways to go:

* Don't define user-defined exceptions, raise built-in exceptions only
* Don't derive from a base ArrowException -- which is there to help indicate that the error arose in C++ code
* Derive from base Exception (worst option, IMHO)

adding @jreback for more opinions (I did this at his asking)